halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/spinnaker/v1/profile/AwsCredentialsProfileFactoryBuilder.java

```diff
@@ -40,12 +46,44 @@
 
   @Setter private String profileName = "default";
 
-  @Setter private String accessKeyId;
+  @Setter(AccessLevel.NONE)
+  private String accessKeyId;
 
-  @Setter private String secretAccessKey;
+  @Setter(AccessLevel.NONE)
+  private String secretAccessKey;
 
   @Setter private SpinnakerArtifact artifact;
 
+  public AwsCredentialsProfileFactoryBuilder setAccessKeyId(String accessKeyId) {
+    if (StringUtils.isEmpty(accessKeyId)) {
+      try {
+        this.accessKeyId =
+            DefaultAWSCredentialsProviderChain.getInstance().getCredentials().getAWSAccessKeyId();
+      } catch (SdkClientException e) {
+        throw new HalException(
+            Problem.Severity.FATAL, "Fail to set aws access key: " + e.getMessage(), e);
+      }
+    } else {
+      this.accessKeyId = accessKeyId;
+    }
+    return this;
+  }
+
+  public AwsCredentialsProfileFactoryBuilder setSecretAccessKey(String secretAccessKey) {
+    if (StringUtils.isEmpty(secretAccessKey)) {
+      try {
+        this.secretAccessKey =
+            DefaultAWSCredentialsProviderChain.getInstance().getCredentials().getAWSSecretKey();
```

Comment:
By separating out the access key and secret key setters, there is a race condition here that the getCredentials result may have changed.